### DISCUSSION TITLE
使用下载 ch_ppstructure_openatom_SLANetv2_infer 模型推理结果存在问题

### DISCUSSION THREAD

**Discussion by TongHengcheng** [PaddlePaddle/PaddleOCR#13299] [Q&A]
#### 问题描述 / Problem Description
我使用下载的ch_ppstructure_openatom_SLANetv2_infer 进行表格识别结果存在问题。

#### 运行环境 / Runtime Environment
- OS: ubuntu 18.4
- Paddle: 2.5.1
- PaddleOCR: 2.8.1

#### 复现代码 / Reproduction Code
det_model_path = "inference/ch_PP-OCRv4_det_infer"
rec_model_path = "inference/ch_PP-OCRv4_rec_infer" 

structure = PPStructure(det_model_dir=det_model_path, 
                        rec_model_dir=rec_model_path, 
                        table_model_dir="inference/ch_ppstructure_openatom_SLANetv2_infer", 
                        #table_model_dir="inference-2.5/ch_ppstructure_mobile_v2.0_SLANet_infer",
                        layout_model_dir="inference-2.5/picodet_lcnet_x1_0_fgd_layout_cdla_infer",
                        table_max_len=512,
                        use_gpu=True,
                        enable_mkldnn=False,
                        benchmark=True
                       )
img_path = "doc/imgs/00018069.jpg"
result = structure(img_path)

#### 完整报错 / Complete Error Message
代码不报错，但结果存在问题，结果如下：
'res': {'cell_bbox': [[166.86669921875, 249.40704345703125, 320.6014404296875, 262.48443603515625]], 'html': '<html><body><table><thead></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody><tr></tr></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody></thead> colspan="20" colspan="20" colspan="20" colspan="20"</tbody><td rowspan="2"</table></body></html>'}

如果使用之前的模型 ch_ppstructure_mobile_v2.0_SLANet_infer，同一个测试数据，结果如下：
'res': {'cell_bbox': [[1.0978889465332031, 0.8825703859329224, 76.5923080444336, 0.9150521755218506, 75.0347671508789, 17.550968170166016, 1.0387251377105713, 17.520965576171875], [67.65865325927734, 0.9864338040351868, 223.51095581054688, 1.0127183198928833, 223.1466064453125, 17.83234214782715, 67.16850280761719, 17.996631622314453], [209.0164794921875, 1.1406828165054321, 289.5438537597656, 1.1863195896148682, 288.6438903808594, 19.356563568115234, 207.0304718017578, 19.37529182434082], [288.12005615234375, 1.414181113243103, 360.0342102050781, 1.4786641597747803, 359.3028259277344, 18.83069610595703, 287.2116394042969, 18.748897552490234], [353.6554260253906, 1.4775595664978027, 428.2085266113281, 1.5463005304336548, 428.1869201660156, 19.91768455505371, 352.8670349121094, 19.9057674407959], [1.4729132652282715, 18.523269653320312, 75.81785583496094, 18.643465042114258, 76.2441177368164, 37.03304672241211, 1.448192834854126, 36.60068130493164], [68.29133605957031, 18.73381233215332, 218.0550079345703, 19.0098876953125, 217.19912719726562, 38.346099853515625, 67.70269012451172, 37.83794021606445], [208.21630859375, 18.762008666992188, 285.88433837890625, 19.10198974609375, 285.5247802734375, 37.674560546875, 207.4384307861328, 37.2264518737793], [283.7080383300781, 19.299339294433594, 356.90692138671875, 19.608985900878906, 356.1598205566406, 37.7532958984375, 282.74993896484375, 37.37887191772461], [356.72503662109375, 18.91582679748535, 428.1366271972656, 19.27887535095215, 428.12176513671875, 37.1911506652832, 356.4380798339844, 36.811561584472656], [1.3411612510681152, 37.94234848022461, 75.23098754882812, 38.29766082763672, 75.3946533203125, 56.64678192138672, 1.3212335109710693, 56.10194778442383], [68.68881225585938, 38.188194274902344, 216.11753845214844, 38.64340591430664, 215.096435546875, 56.66056823730469, 68.09590148925781, 56.019805908203125], [207.56210327148438, 38.1334228515625, 285.0819396972656, 38.585113525390625, 284.4250793457031, 57.03932571411133, 206.7474822998047, 56.562129974365234], [282.0023498535156, 38.538299560546875, 356.7273254394531, 38.97842788696289, 356.0877990722656, 57.027896881103516, 281.3014831542969, 56.549625396728516], [357.4175720214844, 38.5986328125, 427.8137512207031, 39.146358489990234, 427.7963562011719, 56.462860107421875, 357.09765625, 55.87733840942383], [1.5121046304702759, 56.3785514831543, 71.92347717285156, 56.811763763427734, 71.9145278930664, 75.25074005126953, 1.482161521911621, 74.84501647949219], [68.65770721435547, 55.98960494995117, 215.76931762695312, 56.441585540771484, 214.68687438964844, 75.11333465576172, 67.8143081665039, 74.58183288574219], [209.91160583496094, 56.095821380615234, 285.5467529296875, 56.55732727050781, 284.5849914550781, 75.40933990478516, 208.56788635253906, 75.03105926513672], [282.1408386230469, 56.4632453918457, 355.57208251953125, 56.91359329223633, 354.89691162109375, 75.45304107666016, 281.2154846191406, 75.05762481689453], [355.109619140625, 56.73328399658203, 427.71722412109375, 57.4288444519043, 427.69952392578125, 75.09041595458984, 354.9029846191406, 74.36115264892578], [1.7628751993179321, 74.54713439941406, 73.38774108886719, 75.31402587890625, 73.15673065185547, 93.92074584960938, 1.719425916671753, 93.2000732421875], [68.7206039428711, 74.03260040283203, 217.31161499023438, 74.66907501220703, 216.21438598632812, 93.86514282226562, 67.77685546875, 93.25828552246094], [211.12374877929688, 74.08153533935547, 284.928955078125, 74.70024108886719, 283.8206787109375, 94.2997817993164, 209.58419799804688, 93.84551239013672], [282.4198913574219, 74.23876190185547, 355.88037109375, 74.89212799072266, 355.1019592285156, 94.45155334472656, 281.4182434082031, 93.93331146240234], [355.5898742675781, 74.44037628173828, 428.0950012207031, 75.21498107910156, 428.08001708984375, 94.08512115478516, 355.3703918457031, 93.33843994140625], [1.2020732164382935, 93.99455261230469, 72.00303649902344, 94.64751434326172, 72.12586212158203, 113.36302185058594, 1.1822236776351929, 112.71121215820312], [68.48028564453125, 93.53619384765625, 219.72308349609375, 94.21339416503906, 218.8724822998047, 113.59127044677734, 67.79912567138672, 113.0061264038086], [211.78273010253906, 93.56269073486328, 283.4164733886719, 94.2602767944336, 282.5839538574219, 114.27104187011719, 210.75279235839844, 113.7629165649414], [282.51611328125, 93.4762191772461, 355.3823547363281, 94.05813598632812, 354.6811828613281, 113.77680206298828, 281.8313293457031, 113.28302001953125], [358.4176940917969, 93.87555694580078, 427.8598327636719, 94.64068603515625, 427.8460693359375, 113.25995635986328, 358.3985595703125, 112.58428955078125], [1.3357175588607788, 112.01036834716797, 69.82136535644531, 112.77157592773438, 69.89948272705078, 132.35116577148438, 1.3162360191345215, 131.65765380859375], [69.2831802368164, 111.89864349365234, 219.048583984375, 112.70639038085938, 218.17474365234375, 132.5646514892578, 68.59800720214844, 131.8388671875], [213.75692749023438, 112.08163452148438, 284.2778015136719, 112.7608642578125, 283.2928771972656, 132.79031372070312, 212.61549377441406, 132.3220977783203], [282.8636779785156, 112.03659057617188, 354.96221923828125, 112.65286254882812, 354.1809387207031, 132.192626953125, 281.94854736328125, 131.71261596679688], [358.6627197265625, 112.54424285888672, 427.8909606933594, 113.36363983154297, 427.8791198730469, 132.2318572998047, 358.7401123046875, 131.50038146972656], [1.56724214553833, 130.98175048828125, 70.09408569335938, 131.97134399414062, 70.11153411865234, 151.61093139648438, 1.539211630821228, 150.69163513183594], [68.87516784667969, 131.2289581298828, 219.828125, 132.22926330566406, 219.041015625, 151.71133422851562, 68.12822723388672, 150.8408966064453], [215.18121337890625, 130.99053955078125, 283.6591796875, 131.8169403076172, 282.6794128417969, 151.78565979003906, 213.97250366210938, 151.21942138671875], [284.3939208984375, 130.8972930908203, 355.1816101074219, 131.6612548828125, 354.35595703125, 151.43643188476562, 283.37255859375, 150.8695068359375], [359.5212707519531, 131.2664337158203, 427.9620361328125, 132.2101287841797, 427.95123291015625, 151.4951629638672, 359.6144714355469, 150.67050170898438], [1.3200863599777222, 150.40225219726562, 68.9893569946289, 151.3384552001953, 69.09989166259766, 170.5982208251953, 1.299195647239685, 169.7378692626953], [67.31510162353516, 151.2447967529297, 221.1134033203125, 152.2251739501953, 220.36380004882812, 171.2177734375, 66.63763427734375, 170.35545349121094], [215.6154327392578, 150.87879943847656, 283.0735778808594, 151.70272827148438, 282.22662353515625, 171.53585815429688, 214.52342224121094, 170.9358367919922], [283.337158203125, 150.7211151123047, 356.04718017578125, 151.49119567871094, 355.31488037109375, 170.92703247070312, 282.4297790527344, 170.33779907226562], [357.15521240234375, 150.4238739013672, 427.62237548828125, 151.52297973632812, 427.6129150390625, 169.7498016357422, 357.3902282714844, 168.80532836914062], [1.373839259147644, 169.88116455078125, 70.6731185913086, 170.83738708496094, 70.65274810791016, 190.5069580078125, 1.3469173908233643, 189.65338134765625], [68.0196762084961, 170.40579223632812, 222.20606994628906, 171.44259643554688, 221.27960205078125, 190.51319885253906, 67.18135070800781, 189.65008544921875], [215.86448669433594, 170.08950805664062, 283.7099609375, 170.8846435546875, 282.8126220703125, 190.83901977539062, 214.58872985839844, 190.26776123046875], [283.4422912597656, 170.23654174804688, 356.57867431640625, 170.9662322998047, 355.8916015625, 190.4249725341797, 282.50152587890625, 189.86541748046875], [356.969482421875, 170.388916015625, 427.8148193359375, 171.37066650390625, 427.8085632324219, 190.0047149658203, 357.2325134277344, 189.17637634277344], [1.5375503301620483, 189.05311584472656, 71.8197250366211, 190.18984985351562, 71.69197845458984, 209.23182678222656, 1.5011322498321533, 208.31381225585938], [69.35822296142578, 189.2758331298828, 221.78170776367188, 190.3798065185547, 221.04269409179688, 208.96231079101562, 68.4591293334961, 208.14878845214844], [216.22280883789062, 188.82582092285156, 282.7109680175781, 189.70411682128906, 281.8617858886719, 209.2437744140625, 214.8832550048828, 208.67544555664062], [284.1025085449219, 189.0878143310547, 356.21185302734375, 189.90025329589844, 355.5592041015625, 209.17845153808594, 283.08221435546875, 208.63014221191406], [357.4132080078125, 189.37734985351562, 427.6773986816406, 190.45172119140625, 427.67193603515625, 208.7923126220703, 357.6366882324219, 207.97215270996094], [1.240782380104065, 208.1547088623047, 69.82386779785156, 208.93296813964844, 69.82485961914062, 226.59825134277344, 1.2159147262573242, 226.00830078125], [71.7625961303711, 208.61892700195312, 218.57310485839844, 209.4735565185547, 217.73109436035156, 226.63404846191406, 70.74981689453125, 226.04782104492188], [216.1261444091797, 208.24728393554688, 282.5186462402344, 208.9425048828125, 281.74945068359375, 227.70120239257812, 214.7349090576172, 227.27757263183594], [285.3962707519531, 208.4828643798828, 355.1301574707031, 209.06883239746094, 354.5640563964844, 227.97561645507812, 284.39691162109375, 227.6049346923828], [357.03424072265625, 208.652099609375, 427.4294738769531, 209.39247131347656, 427.4260559082031, 226.7340087890625, 357.24560546875, 226.22935485839844], [1.6731010675430298, 227.42051696777344, 71.68177032470703, 228.0846405029297, 71.41840362548828, 247.62901306152344, 1.6369162797927856, 247.2963409423828], [72.3019790649414, 227.96832275390625, 215.9398193359375, 228.57192993164062, 214.97642517089844, 247.9178009033203, 71.0824966430664, 247.64321899414062], [216.5760040283203, 227.95458984375, 284.6583557128906, 228.4362335205078, 283.6848449707031, 248.36083984375, 214.70387268066406, 248.18173217773438], [285.2228088378906, 228.372314453125, 356.8787536621094, 228.76959228515625, 356.4002685546875, 249.96827697753906, 284.2174072265625, 249.8223114013672], [357.24810791015625, 228.8785858154297, 427.39117431640625, 229.22666931152344, 427.39166259765625, 248.1777801513672, 357.6036682128906, 248.03695678710938], [2.1210246086120605, 246.93978881835938, 74.1912841796875, 247.37991333007812, 73.0062484741211, 265.40460205078125, 2.0525784492492676, 265.37799072265625], [71.15603637695312, 247.6543426513672, 214.67825317382812, 247.84878540039062, 213.62644958496094, 265.3997497558594, 69.9049301147461, 265.3854675292969], [215.19876098632812, 247.77667236328125, 287.04327392578125, 247.85569763183594, 285.2253112792969, 265.37445068359375, 212.40321350097656, 265.36566162109375], [285.2870178222656, 247.6883087158203, 357.1719970703125, 247.75335693359375, 356.36590576171875, 265.46978759765625, 283.57354736328125, 265.4630126953125], [358.0940856933594, 247.50942993164062, 427.03631591796875, 247.52090454101562, 426.9951171875, 265.3313903808594, 357.3314208984375, 265.3245544433594]], 'html': '<html><body><table><tbody><tr><td>代号</td><td>项目</td><td>结果</td><td>参考值</td><td>单位</td></tr><tr><td>ALT</td><td>谷丙转氨酶</td><td>25.6</td><td>0--40</td><td>U/L</td></tr><tr><td>TBIL</td><td>总胆红素</td><td>11.2</td><td>&lt;20</td><td>umol/L</td></tr><tr><td>DBIL</td><td>直接胆红素</td><td>3.3</td><td>0-7</td><td>umol/L</td></tr><tr><td>IBIL</td><td>间接胆红素</td><td>7.9</td><td>1.5--15</td><td>umol/L</td></tr><tr><td>TP</td><td>总蛋白</td><td>58.9</td><td>60--80</td><td>g/L</td></tr><tr><td>ALB</td><td>白蛋白</td><td>35.1</td><td>33--55</td><td>g/L</td></tr><tr><td>GLO</td><td>球蛋白</td><td>23.8</td><td>20--30</td><td>g/L</td></tr><tr><td>A/G</td><td>白球比</td><td>1.5</td><td>1.5--2.5</td><td></td></tr><tr><td>ALP</td><td>碱性磷酶</td><td>93</td><td>15--112</td><td>IU/L</td></tr><tr><td>GGT</td><td>谷氨酰转肽酶</td><td>14.3</td><td>&lt;50</td><td>U/L</td></tr><tr><td>AST</td><td>谷草转氨酶</td><td>16.3</td><td>8--40</td><td>U/L</td></tr><tr><td>LDH</td><td>乳酸脱氢酶</td><td>167</td><td>114--240</td><td>U/L</td></tr><tr><td>ADA</td><td>腺甘脱氨酶</td><td>12.6</td><td>4--24</td><td>U/L</td></tr></tbody></table></body></html>'}

**Comment by SWHL** [comment]
模型本身问题，这种一般需要基于自己数据微调模型了

**Comment by invictuszhao** [comment]
尝试用训练模型预测结果，对比和推理模型的结果
